catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveProxyPlugin.java

```diff
@@ -39,28 +39,36 @@ public Object doAs(
       Principal principal, Executable<Object, Exception> action, Map<String, String> properties)
       throws Throwable {
     try {
-      UserGroupInformation proxyUser =
-          UserGroupInformation.createProxyUser(
-              PrincipalUtils.getCurrentPrincipal().getName(), currentUser);
+      UserGroupInformation proxyUser;
 
-      if (UserGroupInformation.isSecurityEnabled()) {
-        if (ops != null) {
+      if (UserGroupInformation.isSecurityEnabled() && ops != null) {
 
-          String token =
-              ops.getClientPool()
-                  .run(
-                      client -> {
-                        return client.getDelegationToken(
-                            currentUser.getUserName(),
-                            PrincipalUtils.getCurrentPrincipal().getName());
-                      });
-
-          Token<DelegationTokenIdentifier> delegationToken = new Token<DelegationTokenIdentifier>();
-          delegationToken.decodeFromUrlString(token);
-          delegationToken.setService(
-              new Text(ops.getHiveConf().getVar(HiveConf.ConfVars.METASTORE_TOKEN_SIGNATURE)));
-          proxyUser.addToken(delegationToken);
+        String kerberosPrincipalName = principal.getName();
+        if (!kerberosPrincipalName.contains("@")) {
+          kerberosPrincipalName =
+              String.format("%s@%s", kerberosPrincipalName, ops.getKerberosRealm());
```

Comment:
Can you explain more about the changes here?

Reply:
Fixed.